### PR TITLE
Replace Math.min() with minOf

### PR DESCRIPTION
**Pull request by TangHuaiZhe**
- Replace `Math.min()` with `minOf`
- Use `0L` replace `0` when compared with a long var

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/internal/http/CallServerInterceptor.kt [hunk 1]
```diff
@@ -116,7 +116,7 @@ class CallServerInterceptor(private val forWebSocket: Boolean) : Interceptor {
         "close".equals(response.header("Connection"), ignoreCase = true)) {
       exchange.noNewExchangesOnConnection()
     }
-    if ((code == 204 || code == 205) && response.body()?.contentLength() ?: -1 > 0) {
+    if ((code == 204 || code == 205) && response.body()?.contentLength() ?: -1 > 0L) {
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/internal/http/CallServerInterceptor.kt]: -1L?

File: okhttp/src/main/java/okhttp3/internal/http2/Http2Writer.kt [hunk 2]
```diff
@@ -100,7 +100,7 @@ class Http2Writer(
     hpackWriter.writeHeaders(requestHeaders)
 
     val byteCount = hpackBuffer.size
-    val length = Math.min((maxFrameSize - 4).toLong(), byteCount).toInt()
+    val length = minOf((maxFrameSize - 4).toLong(), byteCount).toInt()
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/internal/http2/Http2Writer.kt]: the toLong seems weird here given there is toInt, is there better way to express this?

Comment (TangHuaiZhe) [reply okhttp/src/main/java/okhttp3/internal/http2/Http2Writer.kt]: How about` maxFrameSize - 4L`?

File: okhttp/src/main/java/okhttp3/internal/http/CallServerInterceptor.kt (hunk 1 above)
Comment (TangHuaiZhe) [reply okhttp/src/main/java/okhttp3/internal/http/CallServerInterceptor.kt]: Fixed.

File: mockwebserver/src/main/java/okhttp3/mockwebserver/MockWebServer.kt [hunk 3]
```diff
@@ -659,7 +658,7 @@ class MockWebServer : ExternalResource(), Closeable {
     val chunkSizes = ArrayList<Int>()
     val policy = dispatcher.peek()
     if (contentLength != -1L) {
-      hasBody = contentLength > 0
+      hasBody = contentLength > 0L
```
Comment (swankjesse) [inline mockwebserver/src/main/java/okhttp3/mockwebserver/MockWebServer.kt]: thanks!

File: okhttp/src/main/java/okhttp3/internal/http2/Http2Writer.kt [hunk 4]
```diff
@@ -100,7 +100,7 @@ class Http2Writer(
     hpackWriter.writeHeaders(requestHeaders)
 
     val byteCount = hpackBuffer.size
-    val length = Math.min((maxFrameSize - 4).toLong(), byteCount).toInt()
+    val length = minOf(maxFrameSize - 4L, byteCount).toInt()
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/internal/http2/Http2Writer.kt]: nice!

**Pull request merged**